### PR TITLE
opening setup

### PR DESCRIPTION
**Pull request by i-walker**
## Issue
[#491](https://github.com/TUM-Dev/Campus-Android/issues/491)

Still in WIP but I need review along the way. 

Can I swap the old Javafiles with a kotlin implementation in component/ui/openinghour

### CODE REVIEW

File: app/src/main/java/de/tum/in/tumcampusapp/api/app/TUMCabeAPIService.java [hunk 1]
```diff
@@ -209,6 +211,10 @@
 
     // TICKET SALE
 
+    // openingHours\
+    @GET("Api/openingtimes")
+    Observable<OpeningHours.Result> displayCategory(@Query("category") String category);
```
Comment (kordianbruck) [inline app/src/main/java/de/tum/in/tumcampusapp/api/app/TUMCabeAPIService.java]: There is not `category` filter on this endpoint. Please just download the whole file once and do any filtering on the clientside. Cache should be efficient enough, to not make this a problem.

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/openinghour/OpeningApi.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                    .addCallAdapterFactory(
+                            RxJava2CallAdapterFactory.create())
+                    .addConverterFactory(
+                            GsonConverterFactory.create())
+                    .baseUrl(BASE_URL)
+                    .build()
+
+            return retrofit.create(OpeningApi::class.java)
+        }
+    }
```
Comment (kordianbruck) [inline app/src/main/java/de/tum/in/tumcampusapp/component/ui/openinghour/OpeningApi.kt]: You don't need to build your own API here - we already have all of this.

Comment (i-walker) [reply app/src/main/java/de/tum/in/tumcampusapp/component/ui/openinghour/OpeningApi.kt]: Kannst du mir in der Zukunft ein Link Schicken, wenn du ueber eine File redest.

Comment (i-walker) [reply app/src/main/java/de/tum/in/tumcampusapp/component/ui/openinghour/OpeningApi.kt]: [But this is not the Endpoint you created](https://github.com/i-walker/Campus-Android/blob/e65e242c65ef7d479910547694338c1e16ac9e7e/app/src/main/java/de/tum/in/tumcampusapp/api/app/TUMCabeClient.java#L74)

Comment (kordianbruck) [reply app/src/main/java/de/tum/in/tumcampusapp/component/ui/openinghour/OpeningApi.kt]: Retrofit generates all paths for all interface defintions. You don't need to create one retrofit client per endpoint. You can simply use the already initialized instance variable (singleton pattern): https://github.com/TUM-Dev/Campus-Android/blob/master/app/src/main/java/de/tum/in/tumcampusapp/api/app/TUMCabeClient.java#L103

**Pull request closed without merging**